java/ql/test/stubs/google-android-9.0.0/android/content/Context.java

```diff
@@ -0,0 +1,816 @@
+     *             the root of the files directory or one of the following constants
+     *             for a subdirectory:
+     *             {@link android.os.Environment#DIRECTORY_MUSIC},
+     *             {@link android.os.Environment#DIRECTORY_PODCASTS},
+     *             {@link android.os.Environment#DIRECTORY_RINGTONES},
+     *             {@link android.os.Environment#DIRECTORY_ALARMS},
+     *             {@link android.os.Environment#DIRECTORY_NOTIFICATIONS},
+     *             {@link android.os.Environment#DIRECTORY_PICTURES}, or
+     *             {@link android.os.Environment#DIRECTORY_MOVIES}.
+     * @return the absolute paths to application-specific directories. Some
+     *         individual paths may be {@code null} if that shared storage is not
+     *         currently available. The first path returned is the same as
+     *         {@link #getExternalFilesDir(String)}.
+     * @see #getExternalFilesDir(String)
+     * @see Environment#getExternalStorageState(File)
+     * @see Environment#isExternalStorageEmulated(File)
+     * @see Environment#isExternalStorageRemovable(File)
+     */
+    public abstract File[] getExternalFilesDirs(String type);
+
+    /**
+     * Return the primary shared/external storage directory where this application's
+     * OBB files (if there are any) can be found. Note if the application does not
+     * have any OBB files, this directory may not exist.
+     * <p>
+     * This is like {@link #getFilesDir()} in that these files will be deleted when
+     * the application is uninstalled, however there are some important differences:
+     * <ul>
+     * <li>Shared storage may not always be available, since removable media can be
+     * ejected by the user. Media state can be checked using
+     * {@link Environment#getExternalStorageState(File)}.
+     * <li>There is no security enforced with these files. For example, any
+     * application holding
+     * {@link android.Manifest.permission#WRITE_EXTERNAL_STORAGE} can write to these
+     * files.
+     * </ul>
+     * <p>
+     * Starting in {@link android.os.Build.VERSION_CODES#KITKAT}, no permissions are
+     * required to read or write to the path that this method returns. However,
+     * starting from {@link android.os.Build.VERSION_CODES#M}, to read the OBB
+     * expansion files, you must declare the
+     * {@link android.Manifest.permission#READ_EXTERNAL_STORAGE} permission in the
+     * app manifest and ask for permission at runtime as follows:
+     * </p>
+     * <p>
+     * {@code <uses-permission android:name=
+     * "android.permission.READ_EXTERNAL_STORAGE"
+     * android:maxSdkVersion="23" />}
+     * </p>
+     * <p>
+     * Starting from {@link android.os.Build.VERSION_CODES#N},
+     * {@link android.Manifest.permission#READ_EXTERNAL_STORAGE} permission is not
+     * required, so don’t ask for this permission at runtime. To handle both cases,
```

Comment:
Looks like our CI checks also object to that non-ASCII apostrophe

```suggestion
     * required, so don't ask for this permission at runtime. To handle both cases,
```

Reply:
Thanks @smowton. I've replaced that non-ASCII apostrophe.